src/paas_charm/charm.py

```diff
@@ -164,6 +146,124 @@ def __init__(self, framework: ops.Framework, framework_name: str) -> None:
             self.on[self._workload_config.container_name].pebble_ready, self._on_pebble_ready
         )
 
+    def _init_redis(self, requires: dict[str, RelationMeta]) -> "RedisRequires | None":
+        """Initialize the Redis relation if its required.
+
+        Args:
+            requires: relation requires dictionary from metadata
+
+        Returns:
+            Returns the Redis relation or None
+        """
+        _redis = None
+        if "redis" in requires and requires["redis"].interface_name == "redis":
+            try:
+                _redis = RedisRequires(charm=self, relation_name="redis")
+                self.framework.observe(
+                    self.on.redis_relation_updated, self._on_redis_relation_updated
+                )
+            except NameError:
+                logger.exception(
+                    "Missing charm library,                               "
+                    "please run `charmcraft fetch-lib charms.redis_k8s.v0.redis`"
+                )
```

Comment:
We'll need something like this for all the integrations where the libraries might be missing. Check the formatting of the log message

We also need to introduce a check around https://github.com/canonical/paas-charm/blob/38d7f1e2ea36f252a0a1874cd928c092310435c3/src/paas_charm/charm.py#L359 where the logic should be if the integration is detected but the library is missing, the charm goes into blocked state telling the user to include the library in the charm

Reply:
This is needed for:

* S3
* SAML
* Tracing